integration/test/API.hs

```diff
@@ -0,0 +1,144 @@
+module API where
+
+import Data.Aeson hiding ((.=))
+import qualified Data.Array as Array
+import Data.Default
+import Imports
+import System.Random (randomRIO)
+import TestLib.App
+
+randomEmail :: App String
+randomEmail = liftIO $ do
+  n <- randomRIO (8, 15)
+  u <- replicateM n pick
+  pure $ u <> "@example.com"
+  where
+    chars :: Array.Array Int Char
+    chars = mkArray $ ['A' .. 'Z'] <> ['a' .. 'z'] <> ['0' .. '9']
+
+    mkArray :: [a] -> Array.Array Int a
+    mkArray l = Array.listArray (0, length l - 1) l
+
+    pick :: IO Char
+    pick = do
+      i <- randomRIO (Array.bounds chars)
+      pure (chars Array.! i)
+
+defPassword :: String
+defPassword = "hunter2!"
+
+data CreateUser = CreateUser
+  { email :: Maybe String,
+    password :: Maybe String,
+    name :: Maybe String,
+    team :: Bool
+  }
+
+instance Default CreateUser where
+  def =
+    CreateUser
+      { email = Nothing,
+        password = Nothing,
+        name = Nothing,
+        team = False
+      }
+
+createUser :: CreateUser -> App Response
+createUser cu = do
+  email <- maybe randomEmail pure cu.email
+  let password = fromMaybe defPassword cu.password
+      name = fromMaybe email cu.name
+  req <- baseRequest Brig Unversioned "/i/users"
```

Comment:
I think ideally we should do all the scaffolding via the public api, like god wants us to, and only *inspect* the state of the services via internal api where needed.  I've found situations in the past where a test created a corrupt state that was just plausible enough for one test, and when I added more logic to the test it got pretty ugly.

Of course that's a lot more work for eg. this function, where we'd need to model the entire invitation flow.  Maybe we can agree to allow scaffolding via internal api as a quick hack as long as there is a `TODO: use public api` in the haddocks?

Reply:
Yes, agreed. For inviting users I don't know any alternative however.